assets/js/modules/analytics/components/common/EnableUniversalAnalytics.js

```diff
@@ -102,6 +107,31 @@ export default function EnableUniversalAnalytics( {
 		revertPropertyAndProfileIDs,
 	] );
 
+	const isLookingForMatch = useSelect(
+		( select ) =>
+			! select( MODULES_ANALYTICS ).hasFinishedResolution(
+				'getProperties',
+				[ accountID ]
+			)
+	);
+
+	useEffect( () => {
+		( async () => {
+			if ( ! propertyID ) {
+				const matchedProperty = await findMatchedProperty( accountID );
+				if ( matchedProperty?.id ) {
+					selectProperty( matchedProperty.id );
+				}
+			}
+		} )();
+	}, [
+		isUAEnabled,
+		propertyID,
+		findMatchedProperty,
+		selectProperty,
+		accountID,
+	] );
```

Comment:
This looks much better now, thanks!

A new problem I just noticed is that it seems this selects a UA property even though `isUAEnabled` is falsey. This makes the setup/settings flow enable UA even though the user didn't opt for it.

Can you change this so that a UA property is selected only if UA is enabled?